File: templates/problems/17_math_geometry/06_pow_x_n/pow_x_n_test.go
package pow_x_n

import (
	"math"
	"testing"
)

func TestMyPow(t *testing.T) {
	tests := []struct {
		name     string
		x        float64
		n        int
		expected float64
	}{
		{"positive exponent", 2.0, 10, 1024.0},
		{"negative exponent", 2.0, -2, 0.25},
		{"zero exponent", 5.0, 0, 1.0},
		{"exponent 1", 3.0, 1, 3.0},
		{"fractional base", 2.1, 3, 9.261},
		{"base 1", 1.0, 100, 1.0},
		{"base 0", 0.0, 5, 0.0},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got := myPow(tt.x, tt.n)
			if math.Abs(got-tt.expected) > 1e-3 {
				t.Errorf("myPow(%v, %d) = %v, want %v", tt.x, tt.n, got, tt.expected)
			}
		})
	}
}

File: templates/problems/17_math_geometry/06_pow_x_n/typescript/powXN.test.ts
import { describe, it, expect } from 'vitest';
import { myPow } from './powXN';

describe('PowXN', () => {
    it('positive exponent', () => {
        expect(myPow(2.0, 10)).toBeCloseTo(1024.0, 5);
    });

    it('negative exponent', () => {
        expect(myPow(2.0, -2)).toBeCloseTo(0.25, 5);
    });

    it('zero exponent', () => {
        expect(myPow(5.0, 0)).toBeCloseTo(1.0, 5);
    });

    it('exponent 1', () => {
        expect(myPow(3.0, 1)).toBeCloseTo(3.0, 5);
    });

    it('fractional base', () => {
        expect(myPow(2.1, 3)).toBeCloseTo(9.261, 5);
    });

    it('base 1', () => {
        expect(myPow(1.0, 100)).toBeCloseTo(1.0, 5);
    });

    it('base 0', () => {
        expect(myPow(0.0, 5)).toBeCloseTo(0.0, 5);
    });
});
